src/server/pkg/obj/obj.go

```diff
@@ -4,11 +4,33 @@ import (
 	"io"
 	"time"
 
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/storagegateway"
 	"github.com/cenkalti/backoff"
 	"go.pedge.io/lion/proto"
 	"golang.org/x/net/context"
+	"google.golang.org/api/googleapi"
 )
 
+// isRetryable determines if an error is retryable.
+func isRetryable(err error) bool {
+	switch err := err.(type) {
+	case *googleapi.Error:
+		return err.Code >= 500
+	case awserr.Error:
+		for _, c := range []string{
+			storagegateway.ErrorCodeServiceUnavailable,
+			storagegateway.ErrorCodeInternalError,
+			storagegateway.ErrorCodeGatewayInternalError,
+		} {
+			if c == err.Code() {
+				return true
+			}
+		}
+	}
+	return false
+}
```

Comment:
Could we move this logic into the amazon and google implementations. This file shouldn't have anything specific to either of those implementations. I think we want an `IsRetryable` method on the `Client` interface which will dispatch to the right method.
That way when you implement a new obj store you know that you have to implement the method for your own error types.